### DISCUSSION TITLE
Working tuya with smart lock

### DISCUSSION THREAD

**Discussion by farhad1359** [jasonacox/tinytuya#531] [Q&A]
I'm trying to connect to the tuya smart lock, locally. But  I got the 901 error. How I can resolve this issue?

**Comment by jasonacox** [comment]
Hi @farhad1359 - That [error](https://github.com/jasonacox/tinytuya) indicates it is unable to connect. 

This could be caused by the device not being on the WiFi all the time. If it is battery powered, it will only connect to the WiFi to check in to the Tuya Cloud or to react to events. However, if this is not a battery powered device, I would try:

Scan - Does it find the device?
```bash
python3 -m tinytuya scan 
```

Code 
```python
# add debug to your code - also verify IP address, ID and localkey

tinytuya.set_debug(True)
d = tinytuya.Device('DEVICE_ID_HERE', 'IP_ADDRESS_HERE', 'LOCAL_KEY_HERE', version=3.3)
data = d.status()  
print(data)
```

**Comment by farhad1359** [comment]
Hi @jasonacox ,
Thanks for reply.
Yes, you are right. The device just connect to the WiFi and send the event to the cloud and then disconnect from WiFi.
The scan command can find the device. 
Any other idea how I can get data from this smart lock?

**Reply by jasonacox** [reply]
Getting data from the device directly will likely drain the battery (which is why it disconnects from the WiFi). But if you wanted to, you can probably continually ping the device and once it wakes up, grab the status. I have seen some who are successful at that, but it isn't recommended unless you have an external power supply for the device. And, in some cases, the devices won't cooperate.

An "indirect monitoring" alternative is to set up an external device like a smart plug that doesn't control anything, but set up some automation (via SmartLife, Alexa, ITTT, etc) so that when the lock status changes in the cloud, it turns on or off the smart plug. You can then set up your monitoring to watch the smart plug to detect state of the smart lock. I know it isn't quite as good as direct monitoring but some have tried it and have a lot of success with this method.

Another less desired alternative is to monitor the device via the cloud (see https://github.com/jasonacox/tinytuya?tab=readme-ov-file#tuya-cloud-access). However, the free Tuya IoT account will have limited API requests so you could easily run out if you are monitoring frequently.
